Source/Core/Core/PowerPC/GDBThread.cpp

```diff
@@ -0,0 +1,1274 @@
+    return;                                                                                        \
+  }
+
+#define failr(msg)                                                                                 \
+  {                                                                                                \
+    ERROR_LOG(GDB_THREAD, msg);                                                                    \
+    GDBDeinit();                                                                                   \
+    return 0;                                                                                      \
+  }
+
+#define GDB_MAX_BP 100
+
+#define GDB_STUB_START '$'
+#define GDB_STUB_END '#'
+#define GDB_STUB_ACK '+'
+#define GDB_STUB_NAK '-'
+
+#ifdef _WIN32
+#ifndef MSG_WAITALL
+#define MSG_WAITALL 8
+#endif
+#endif
+
+#define REGISTER_ID(category, index) ((category << 8) | index)
+
+// --------------------------------------------------------------------------------------
+//  GDBThread Implementations
+// --------------------------------------------------------------------------------------
+GDBThread::GDBThread() = default;
+
+GDBThread::~GDBThread()
+{
+  Terminate();
+}
+
+bool GDBThread::Initialize()
+{
+  NOTICE_LOG(GDB_THREAD, "GDB thread: Initialize");
+
+  const SConfig& config = SConfig::GetInstance();
+
+  if (config.iGDBPort <= 0)
+    return false;
+
+  m_server_thread = std::thread(&GDBThread::ThreadFunc, this);
+
+  return true;
+}
+
+void GDBThread::Terminate()
+{
+  NOTICE_LOG(GDB_THREAD, "GDB thread: Terminate");
+
+  OnStop();
+
+  if (m_server_thread.joinable())
+    m_server_thread.join();
+}
+
+void GDBThread::OnStart()
+{
+  NOTICE_LOG(GDB_THREAD, "GDB thread: On Start");
+
+  Common::SetCurrentThreadName("GDBThread");
+}
+
+void GDBThread::OnStop()
+{
+  NOTICE_LOG(GDB_THREAD, "GDB thread: On Stop");
+
+  m_is_running = false;
+
+  GDBSignal(SignalType::Terminate);
+  GDBDeinit();
+}
+
+void GDBThread::OnPause()
+{
+  NOTICE_LOG(GDB_THREAD, "GDB thread: On Pause");
+
+  BreakPoints::TTriggeredBreakPoint triggered_breakpoint =
+      PowerPC::breakpoints.GetBreakpointTriggered();
+
+  if (CPU::State::PowerDown == CPU::GetState())
+  {
+    GDBSignal(SignalType::Terminate);
+    GDBDeinit();
+  }
+  else if (std::get<0>(triggered_breakpoint))
+  {
+    u64 addr = std::get<1>(triggered_breakpoint).value_or(UINT64_MAX);
+    MemCheckCondition cond = std::get<2>(triggered_breakpoint).value_or(MemCheckCondition::Execute);
+
+    GDBSignal(SignalType::Trap, addr, cond);
+  }
+  else
+  {
+    GDBSignal(SignalType::Stop);
+  }
+}
+void GDBThread::OnResume()
+{
+  NOTICE_LOG(GDB_THREAD, "GDB thread: On Resume");
+
+  if (CPU::State::PowerDown == CPU::GetState())
+  {
+    GDBSignal(SignalType::Terminate);
+    GDBDeinit();
+  }
+  else
+  {
+    GDBSignal(SignalType::Continue);
+  }
+}
+
+void GDBThread::UpdateState(CPU::State current_state)
+{
+  if (current_state != m_previous_state)
```

Comment:
You could invert this check and exit early, to get rid of a level of indentation:
```suggestion
  if (current_state == m_previous_state)
    return;
```

Reply:
True. Will make the change. Thanks.